src/helpers.py

```diff
@@ -65,3 +66,28 @@ def parse_ca_chain(ca_chain_pem: str) -> List[str]:
     if not chain_list:
         raise ValueError("No certificate found in chain file")
     return chain_list
+
+
+def ca_chain_is_valid(ca_chain: List[str]) -> bool:
+    """Returns whether a ca chain is valid.
+
+    It uses the x509 certificate method verify_directly_issued_by.
+    It checks the certificate issuer name matches the issuer subject name and that
+    the certificate is signed by the issuer's private key.
+
+    Args:
+        ca_chain: composed by a list of certificates.
+
+    Returns:
+        whether the ca chain is valid.
+    """
+    if len(ca_chain) < 2:
+        return False
+    for ca_cert, cert in zip(ca_chain, ca_chain[1:]):
+        try:
+            ca_cert_object = x509.load_pem_x509_certificate(ca_cert.encode("utf-8"))
+            cert_object = x509.load_pem_x509_certificate(cert.encode("utf-8"))
+            cert_object.verify_directly_issued_by(ca_cert_object)
+        except (ValueError, TypeError, InvalidSignature):
```

Comment:
We should probably separate the excepts and log different messages for each. I suspect each one is a different reason for this ca chain not to be valid

Reply:
These are the possible exceptions returned by `verify_directly_issued_by` according to https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.verify_directly_issued_by 
I think we can group and log the exception message. What do you think ?